spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialMaxPooling.scala

```diff
@@ -87,6 +87,39 @@ class SpatialMaxPooling[T: ClassTag](
     this
   }
 
+  override def computeOutputShape(inputShape: Shape): Shape = {
+    val input = inputShape.toSingle().toArray
+    require(input.length == 4,
+      "SpatialMaxPooling: " + ErrorInfo.constrainInputAs3DOrBatch)
```

Comment:
We should add the input length into the error message in respect of debugging. Please update the other require error message as well.